src/celeritas/neutron/model/ChipsNeutronElasticModel.cc

```diff
@@ -0,0 +1,299 @@
+        InteractionApplier{ChipsNeutronElasticExecutor{this->host_ref()}});
+    return launch_action(*this, params, state, execute);
+}
+
+//---------------------------------------------------------------------------//
+#if !CELER_USE_DEVICE
+void ChipsNeutronElasticModel::execute(CoreParams const&, CoreStateDevice&) const
+{
+    CELER_NOT_CONFIGURED("CUDA OR HIP");
+}
+#endif
+
+//---------------------------------------------------------------------------//
+/*!
+ * Get the model ID for this model.
+ */
+ActionId ChipsNeutronElasticModel::action_id() const
+{
+    return this->host_ref().ids.action;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct interaction cross section data for a single element.
+ */
+void ChipsNeutronElasticModel::append_xs(ImportPhysicsVector const& inp,
+                                         HostXsData* data) const
+{
+    auto reals = make_builder(&data->reals);
+    GenericGridData micro_xs;
+
+    // Add the tabulated interaction cross section from input
+    micro_xs.grid = reals.insert_back(inp.x.begin(), inp.x.end());
+    micro_xs.value = reals.insert_back(inp.y.begin(), inp.y.end());
+    micro_xs.grid_interp = Interp::linear;
+    micro_xs.value_interp = Interp::linear;
+
+    // Add micro xs data
+    CELER_ASSERT(micro_xs);
+    make_builder(&data->micro_xs).push_back(micro_xs);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct A-dependent coefficients of the CHIPS differential cross section
+ * for a single isotope used in the G4ChipsNeutronElasticXS class.
+ *
+ * \note The parameter coeffs[index] is equivalent to lastPAR[index + 15] used
+ * in G4ChipsNeutronElasticXS::GetChipsCrossSection. The first 15 parameters
+ * of lastPAR are not used for sampling the momentum transfer between neutron
+ * and nucleus (Z > 0, N >= 1).
+ *
+ * M.V. Kossov, “Manual for the CHIPS event generator”, KEK internal report
+ * 2000-17, Feb. 2001 H/R; P.V. Degtyarenko, M.V. Kossov and H.P. Wellisch,
+ * Eur. Phys. J. A8, 217 (2000); P.V. Degtyarenko, M.V. Kossov and H.P.
+ * Wellisch, Eur. Phys. J. A9 (2001).
+ */
+void ChipsNeutronElasticModel::append_coeffs(AtomicMassNumber A,
+                                             HostXsData* data) const
+{
+    ChipsDiffXsCoefficients coeffs;
+
+    real_type a = static_cast<real_type>(A.get());
+
+    real_type sa = std::sqrt(a);
+    real_type asa = a * sa;
+    real_type a2 = ipow<2>(a);
+    real_type a3 = a2 * a;
+    real_type a4 = a3 * a;
+    real_type a5 = a4 * a;
+    real_type a6 = a4 * a2;
+    real_type a7 = a6 * a;
+    real_type a8 = a7 * a;
+    real_type a9 = a8 * a;
+    real_type a10 = a9 * a;
+    real_type a12 = ipow<2>(a6);
+    real_type a14 = ipow<2>(a7);
+    real_type a16 = ipow<2>(a8);
+    real_type a32 = ipow<2>(a16);
+
+    // The differential cross-section is parameterized separately for
+    // A>6 & A<7
+    if (a < 6.5)
```

Comment:
I can tell where this statement came from 😂

Try again:
```suggestion
    if (amass <= AtomicMassNumber(6))
```

Reply:
Changed as suggested.